core/src/main/java/org/bf2/srs/fleetmanager/execution/impl/workers/provision/ScheduleRegistryWorker.java

```diff
@@ -85,10 +87,11 @@ public void execute(Task aTask, WorkerContext ctl) throws RegistryStorageConflic
         // NOTE: Failure point 3
         storage.createOrUpdateRegistry(registry);
 
-        ctl.delay(() -> tasks.submit(ProvisionRegistryTenantTask.builder().registryId(registry.getId()).build()));
```

Comment:
The delay feature must be used to allow the current transaction to be finished before running the downstream task, otherwise the downstream task could see the old status value. Same in the new task.

Reply:
> This looks good, however there is an issue with reporting errors from async tasks. For example the org being out of available subscriptions. I was thinking on adding an error field into the Registry schema component, or more generally an extended status. This must be correctly handled in the UI as well.

Yes, that is one of the things I want to polish for sure.

Reply:
> The delay feature must be used to allow the current transaction to be finished before running the downstream task, otherwise the downstream task could see the old status value. Same in the new task.

While I see what you're saying here (and agree), there are other ways of ensuring that the transaction is done before starting the other task. Is any other option also valid or there is something I'm missing here (possible)?

Reply:
Nevermind, I have fixed that part now. Back to work on the error reporting side.